src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
```

Comment:
How about using some underscore utilities to make this code more concise? For example:

```javascript
  //if the element doesn't belong to document DOM observe DOM to detect
  //when it's attached
  var observer = null;
  if (!document.body.contains(el) && reactElement) {
    observer = new MutationObserver(mutations => {
      var updateSize = !_.chain(mutations)
        .filter(m => m.type === 'childList' && _.contains(m.addedNodes, el))
        .isEmpty()
        .value();
      reactElement.setState({updateSize});
    });

    // start observing document
    observer.observe(document, {childList: true});
  }

/* ... */

      // disconnect observer if it was created
      if (observer) {
        observer.disconnect();
      }
```